src/MultiTransfer.sol

```diff
@@ -0,0 +1,96 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+/**
+ * @title MultiTransfer
+ * @dev This contract allows the admin to send ETH to multiple recipients in a single transaction.
+ *      If a transfer fails, it's skipped.
+ *      Failed transfers' amounts are accumulated and sent back to the admin.
+ *      Only an admin can initiate transfers and withdraw funds.
+ */
+contract MultiTransfer is AccessControl {
+    /// @notice Emitted when a transfer is successful.
+    event Transfer(address indexed to, uint256 amount);
+
+    /// @notice Emitted when a transfer fails.
+    event TransferFailed(address indexed to, uint256 amount);
+
+    /// @notice Emitted when the admin withdraws funds.
+    event Withdrawn(address indexed admin, uint256 amount);
+
+    /**
+     * @dev Sets the original `msg.sender` as the admin.
+     */
+    constructor() {
+        _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+    }
+
+    /**
+     * @notice Transfers the specified amounts of ETH to the specified recipients.
+     * @param recipients Array of recipient addresses.
+     * @param amounts Array of amounts of ETH to send.
+     * @dev The arrays `recipients` and `amounts` must have the same length.
+     *      If a transfer fails, the transfer is skipped.
+     *      Failed transfers' amounts are accumulated and sent back to the admin.
+     *      Can only be called by the admin.
+     */
+    function multiTransfer(
+        address payable[] memory recipients,
+        uint256[] memory amounts
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(
+            recipients.length == amounts.length,
+            "Recipients and amounts length mismatch"
+        );
+
+        uint256 refundAmount = 0;
+
+        for (uint256 i = 0; i < recipients.length; i++) {
+            uint256 amount = amounts[i];
+            if (!attemptTransfer(recipients[i], amount)) {
+                emit TransferFailed(recipients[i], amount);
+                refundAmount += amount;
+            }
+        }
+
+        if (refundAmount > 0) {
+            bool sent = payable(msg.sender).send(refundAmount);
```

Comment:
Why does the refund amount not stay in the contract?

Reply:
The assumption here is that if a transfer fails on a given address, it will always fail so best to return the funds to the original IE contract immediately where it can be repurposed.